common/lib/types/protocolmessage.js

```diff
@@ -128,7 +154,20 @@ var ProtocolMessage = (function() {
 			result += '; flags=' + Utils.arrFilter(flagNames, function(flag) {
 				return msg.hasFlag(flag);
 			}).join(',');
-
+		if(msg.params) {
+			var stringifiedParams = '';
+			for (var key in msg.params) {
```

Comment:
Surprised this pattern of `for (var key in msg.params) {  if (Object.prototype.hasOwnProperty.call(msg.params, key) && msg.params[key])` doesn't have a Utils helper.  Seems to be used quite widely and thus adds to the overall library weight.

Reply:
@SimonWoolf wdyt?

Reply:
Refactored in https://github.com/ably/ably-js/pull/613/commits/b727b65b35ef4d05a66bc1902d886d1ab1286ce6